monkey/infection_monkey/i_puppet/i_fingerprinter.py

```diff
@@ -0,0 +1,27 @@
+from abc import abstractmethod
+from typing import Dict
+
+from . import FingerprintData, PingScanData, PortScanData
+
+
+class IFingerprinter:
+    @abstractmethod
+    def get_host_fingerprint(
+        self,
+        host: str,
+        ping_scan_data: PingScanData,
```

Comment:
Why do we need this param?

Reply:
Wouldn't it be better to simply not invoke the fingerprinter if we couldn't reach the host? I think it should be the responsibility of the caller to decide when to invoke the fingerprinter and when not to, based on ping scan results

Reply:
A host may not respond to ICMP requests but still have TCP ports open.

Reply:
That's what I'm saying, if ICMP is not related to TCP, then why do we pass the results of ICMP?

Reply:
There may be a fingerprinter that uses ICMP data to draw conclusions about the host. For example, the ping scanner right now is looking at TTL to determine OS. That should probably be a fingerprinter. The ping scanner should be responsible for collecting the data. We can have an ICMPFingerprinter that determines the OS.

Other fingerprinters may need to use ICMP data in combination with some other data to draw conclusions. In summary, we provide all available scanning data to the fingerprinters and let them decide how to use that data, since the master can't know what any particular fingerprinter needs.